packages/static-config/src/swc.ts

```diff
@@ -0,0 +1,188 @@
+  return node.type === 'StringLiteral';
+}
+
+function isNumericLiteral(node: Node): node is NumericLiteral {
+  return node.type === 'NumericLiteral';
+}
+
+function isArrayExpression(node: Node): node is ArrayExpression {
+  return node.type === 'ArrayExpression';
+}
+
+function isObjectExpression(node: Node): node is ObjectExpression {
+  return node.type === 'ObjectExpression';
+}
+
+function isKeyValueProperty(node: Node): node is KeyValueProperty {
+  return node.type === 'KeyValueProperty';
+}
+
+export type Value = undefined | null | boolean | string | number | any[] | Record<string, any>;
+export class UnsupportedValueError extends Error {}
+
+function extractValue(node: Node): Value {
+  if (isNullLiteral(node)) {
+    return null;
+  } else if (isBooleanLiteral(node)) {
+    // e.g. true / false
+    return node.value;
+  } else if (isStringLiteral(node)) {
+    // e.g. "abc"
+    return node.value;
+  } else if (isNumericLiteral(node)) {
+    // e.g. 123
+    return node.value;
+  } else if (isIdentifier(node)) {
+    switch (node.value) {
+      case 'undefined':
+        return undefined;
+      default:
+        throw new UnsupportedValueError();
+    }
+  } else if (isArrayExpression(node)) {
+    // e.g. [1, 2, 3]
+    const arr = [];
+    for (const elem of node.elements) {
+      if (elem) {
+        if (elem.spread) {
+          // e.g. [ ...a ]
+          throw new UnsupportedValueError();
+        }
+
+        arr.push(extractValue(elem.expression));
+      } else {
+        // e.g. [1, , 2]
+        //         ^^
+        arr.push(undefined);
+      }
+    }
+    return arr;
+  } else if (isObjectExpression(node)) {
+    // e.g. { a: 1, b: 2 }
+    const obj: Record<string, any> = {};
+    for (const prop of node.properties) {
+      if (!isKeyValueProperty(prop)) {
+        // e.g. { ...a }
+        throw new UnsupportedValueError();
+      }
+
+      let key: string;
+      if (isIdentifier(prop.key)) {
+        // e.g. { a: 1, b: 2 }
+        key = prop.key.value;
+      } else if (isStringLiteral(prop.key)) {
+        // e.g. { "a": 1, "b": 2 }
+        key = prop.key.value;
+      } else {
+        throw new UnsupportedValueError();
+      }
+
+      obj[key] = extractValue(prop.value);
+    }
+
+    return obj;
+  } else {
+    throw new UnsupportedValueError();
+  }
+}
+
+// Extracts the value of an exported const variable named `exportedName`
+// (e.g. "export const config = { runtime: 'edge' }") from swc's AST.
+//
+// The value must be one of (or throws UnsupportedValueError):
+//
+// - string
+// - boolean
+// - number
+// - null
+// - undefined
+// - array containing values listed in this list
+// - object containing values listed in this list
+//
+export function extractExportedConstValue(
+  module: Module,
+  exportedName: string
+): Value | null {
+  for (const moduleItem of module.body) {
+    if (!isExportDeclaration(moduleItem)) {
+      continue;
+    }
+
+    const declration = moduleItem.declaration;
```

Comment:
This has a typo. Perhaps use the following:

```suggestion
    const { declaration } = moduleItem;
```